File: src/main/java/javabot/web/auth/Restricted.java
package javabot.web.auth;

import javabot.web.model.Authority;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Documented
@Retention(RetentionPolicy.RUNTIME)
@Target({ElementType.PARAMETER})
public @interface Restricted {
  Authority[] value() default Authority.ROLE_ADMIN;
}

File: src/main/kotlin/javabot/web/auth/Restricted.kt
package javabot.web.auth

import javabot.web.model.Authority
import javabot.web.model.Authority.ROLE_ADMIN

@MustBeDocumented
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.VALUE_PARAMETER)
annotation class Restricted(vararg val value: Authority = arrayOf(ROLE_ADMIN))
